### DISCUSSION TITLE
Setting Random Seed for all Processes

### DISCUSSION THREAD

**Discussion by Kai-X-Org** [HumanSupportRobot/QandA#102] [Q&A]
Hi,

Is there a way to set random seeds for all the processes going on in the HSR simulation? From the ROS nodes to things happening in the robot's software? Thank you so much!

**Comment by k-takeshita** [comment]
I did a quick search, but it appears there is no way to set a seed.

The HSR's gazebo simulator includes the gazebo_ros's empty_world.launch, which originally does not have an argument to set the seed (although gzserver can be set with --seed).
Additionally, the seed used in the RRT planner for motion planning in HSR is fixed, as it utilizes std::mt19937 without any arguments.

**Reply by Kai-X-Org** [reply]
Got it! Thank you so much! Things are good for me as long as the seed is fixed for the random parts.
